google/cloud/logging_v2/handlers/_monitored_resources.py

```diff
@@ -233,3 +248,40 @@ def detect_resource(project=""):
     else:
         # use generic global resource
         return _create_global_resource(project)
+
+
+def add_environment_labels(resource: Resource, record: logging.LogRecord):
+    """Returns additional labels to be appended on to a LogRecord object based on the
+    local environment. Defaults to an empty dictionary if none apply. This is only to be
+    used for CloudLoggingHandler, as the structured logging daemon already does this.
+
+    Args:
+        resource (google.cloud.logging.Resource): Resource based on the environment
+        record (logging.LogRecord): A LogRecord object representing a log record
+    Returns:
+        Dict[str, str]: New labels to append to the labels of the LogRecord
+    """
+    additional_labels = {}
+
+    def set_item(key, val):
+        if val:
+            additional_labels[key] = val
```

Comment:
This function seems a bit unoptimized for something that has to be run on each log

Can we remove this inner function?

I assume the env vars aren't expected to change during runtime, right? If we can rely on that, maybe we can build the additional_labels dict once at init time, and attach it to each `cloud_run_jobs` log?

Reply:
Looking at it, we can build part of the `additional_labels` dict once at init time, but something like setting the `trace_id` label will have to be done per log record, as that label depends on what the value of `_trace` is in each `LogRecord`.

Reply:
The env vars aren't expected to change during runtime, but I've also refactored the GAE `trace_id` label code into this change as well. Does that change during runtime?